### PR TITLE
Tabs: Don't allow navigating to hidden tabs

### PR DESCRIPTION
**Pull request by scottgonzalez**
Fixes #15013

### CODE REVIEW

File: ui/widgets/tabs.js [hunk 1]
```diff
@@ -270,7 +271,19 @@ $.widget( "ui.tabs", {
 			return index;
 		}
 
-		while ( $.inArray( constrain(), this.options.disabled ) !== -1 ) {
+		function canActivate( index ) {
+			if ( $.inArray( constrain(), that.options.disabled ) !== -1 ) {
+				return false;
+			}
+
+			if ( that.tabs.eq( index ).is( ":hidden" ) ) {
```
Comment (jzaefferer) [inline ui/widgets/tabs.js]: How about...
```
return that.tabs.eq( index ).is( ":visible" )
```

Comment (scottgonzalez) [inline ui/widgets/tabs.js]: When there are multiple failure cases, I generally like to list them out individually as failure cases, then return `true` at the end, but I'm fine making this change.

**Comment by jzaefferer** [comment]
Besides that one thing, 👍

File: ui/widgets/tabs.js (hunk 1 above)
Comment (arschmitz) [inline ui/widgets/tabs.js]: Just a thought but the `:visible` and `:hidden` selectors are pretty slow and not 100% reliable. We don't document that hidden tabs are automatically not navigable. Would it perhaps be better to leave this change out and document that you should mark hidden tabs as disabled?

Comment (scottgonzalez) [inline ui/widgets/tabs.js]: Actually, for some reason switching it causes the tests to fail.

Comment (scottgonzalez) [inline ui/widgets/tabs.js]: > We don't document that hidden tabs are automatically not navigable.

Why would anyone ever expect a hidden tab to be navigable?

Comment (arschmitz) [inline ui/widgets/tabs.js]: I don't think they would which is why i suggested documenting that hidden tabs should be marked disabled

Comment (scottgonzalez) [inline ui/widgets/tabs.js]: https://bugs.jqueryui.com/ticket/15013#comment:2

**Comment by scottgonzalez** [comment]
I'm going to close this because this will result in an odd state of tabs not being marked as disabled, but also not being able to be activated properly. Instead, we should implement the hidden detection inside `refresh()` and set those tabs as disabled.

**Pull request closed without merging**